c7n/resources/s3.py

```diff
@@ -789,6 +1029,212 @@ def process_bucket(self, func, bucket, topic, account_id, session_factory):
+        if glacier_days_in_policy and (bucket_region in glacier_supported_regions):
+            rem_lifecycle.set_glacier_transition_days(glacier_days_in_policy)
+        if multipart_days_in_policy:
+            rem_lifecycle.set_stale_multipart_uploads_days(multipart_days_in_policy)
+        if delete_days_in_policy:
+            rem_lifecycle.set_deletes_objects_days(delete_days_in_policy)
+
+        bad_bucket_lifecycle = True
+        rules = {'Rules': []}
+        if 'LifecyclePolicy' in bucket:
+            lifecycle = bucket['LifecyclePolicy']
+            for rule in lifecycle['Rules']:
+                if rule['ID'] == self.data.get('id'):
+                    bad_bucket_lifecycle = False
+
+                    if rule['Status'] == 'Disabled':
+                        bad_bucket_lifecycle = True
+
+                    # Handle lifecycles created using new Amazon web interface
+                    if 'Filter' in rule:
+                        if rule['Filter']['Prefix'] != prefix:
+                            bad_bucket_lifecycle = True
+                    else:
+                        if rule['Prefix'] != prefix:
+                            bad_bucket_lifecycle = True
+
+                    if 'Transitions' in rule:
+                        # Can only have STANDARD_IA and GLACIER transitions
+                        if (len(rule['Transitions']) < 2) and\
+                                ia_days_in_policy and\
+                                glacier_days_in_policy:
+                            bad_bucket_lifecycle = True
+                        else:
+                            for t in rule['Transitions']:
+                                if (t['StorageClass'] == 'STANDARD_IA'):
+                                    if ia_days_in_policy and (t['Days'] != ia_days_in_policy):
+                                        rem_lifecycle.set_ia_transition_days(ia_days_in_policy)
+                                        bad_bucket_lifecycle = True
+                                    else:
+                                        rem_lifecycle.set_ia_transition_days(t['Days'])
+
+                                if (t['StorageClass'] == 'GLACIER'):
+                                    if glacier_days_in_policy and\
+                                            (t['Days'] != glacier_days_in_policy):
+                                        rem_lifecycle.set_glacier_transition_days(
+                                            glacier_days_in_policy
+                                        )
+                                        bad_bucket_lifecycle = True
+                                    else:
+                                        rem_lifecycle.set_glacier_transition_days(t['Days'])
+
+                    elif ia_days_in_policy or glacier_days_in_policy:
+                        bad_bucket_lifecycle = True
+
+                    if 'NoncurrentVersionTransitions' in rule:
```

Comment:
I have added functionality that will allow the user to both look for and configure lifecycles that deal with previous versions of objects.